eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/permission/broadcast/PermissionRequestResult.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.permission.broadcast;
+
+import android.app.Activity;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+
+/**
+ * The data for the {@link Activity#onRequestPermissionsResult(int, String[], int[])} callback.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PermissionRequestResult implements Parcelable
+{
+    public final int requestCode;
```

Comment:
This was deliberately intended as what you call a data bag. The data that was delivered in `Activity.onPermissionRequestResult(int, String[], int[])`, to be able to send them in a broadcast without any interpretation. I didn't want to add any abstraction to this part. I wanted it to be independent from `Permission` and `PermissionStatus`es, and allow any interested party to listen in on this broadcast, like the partner app itself, if it's something useful for them.
So this is intended to be raw data, sort of. Not opinionated, not generalized.
And if it's just raw data, I am not sure that there will be any use for the interface, maybe there will be, I just don't see it.

So if we want to change this, I see 3 options now:

1. Create interface with the same methods for the raw data:
```
interface PermissionResult
{
     int requestCode();
     String[] permissions();
     int[] grantResults();
}
```

2. Add them separately to the broadcast Intent with 3 keys

3. Create some interpreted interface to represent the result, not sure how best to do it, 2 examples:

a.
```
interface PermissionResult
{
     boolean wasInterrupted();
     boolean granted(String permissionName) throws IllegalArgumentExcepttion, IllegalStateException;
}
```
b.
```
interface PermissionResult
{
     boolean wasInterrupted();
    List<String> granted();
    List<String> denied();
}
```

But as I mentioned I would avoid adding another abstraction 'layer' here, so I would go with 1., or maybe 2.
What do you say?